internal/collector/telemetry/process_telemetry.go

```diff
@@ -0,0 +1,105 @@
+package telemetry
+
+import (
+	"context"
+	"os"
+	"runtime"
+	"time"
+
+	"github.com/prometheus/procfs"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+// ProcessMetricsViews is a struct that contains views related to process metrics (cpu, mem, etc)
+type ProcessMetricsViews struct {
+	views []*view.View
+	done  chan struct{}
+}
+
+var mRuntimeAllocMem = stats.Int64("oc.io/process/memory_alloc", "Number of bytes currently allocated in use", "By")
+var viewAllocMem = &view.View{
+	Name:        mRuntimeAllocMem.Name(),
+	Description: mRuntimeAllocMem.Description(),
+	Measure:     mRuntimeAllocMem,
+	Aggregation: view.LastValue(),
+	TagKeys:     []tag.Key{},
+}
+
+var mRuntimeTotalAllocMem = stats.Int64("oc.io/process/total_memory_alloc", "Number of bytes allocated in total", "By")
+var viewTotalAllocMem = &view.View{
+	Name:        mRuntimeTotalAllocMem.Name(),
+	Description: mRuntimeTotalAllocMem.Description(),
+	Measure:     mRuntimeTotalAllocMem,
+	Aggregation: view.Count(),
+	TagKeys:     []tag.Key{},
+}
+
+var mRuntimeSysMem = stats.Int64("oc.io/process/sys_memory_alloc", "Number of bytes given to the process in total", "By")
```

Comment:
memory of the system?

Reply:
Confusingly named by golang, I believe this is actually the amount of memory given to the process that it could allocate